### ISSUE TITLE
Model training failure?

### ISSUE THREAD

**Issue by wirawan0** [Xtra-Computing/thundersvm#126]
I am recently using CIC-IDS-2017 dataset (https://iscxdownloads.cs.unb.ca/iscxdownloads/CIC-IDS-2017/#CI-IDS-2017) to test machine learning algorithm to identify network threats. I am using scikit-learn package + `thundersvmScikit` library. Tried to use SVM on the Tuesday dataset (archive: GeneratedLabelledFlows.zip, filename in the archive: Tuesday-WorkingHours.pcap_ISCX.csv). Attached is a reduced testcase from that dataset by sampling only 5% (~22k records) of the entire dataset. (There has been some massaging to select 76 out of >80 features found in the original CSV file.)

**Issue observed**: When I tried to train a ThunderSVM model on this entire dataset, I ended up getting only 0.02 accuracy score (measured by `sklearn.metrics.accuracy_score`). The same dataset will yield accuracy of `0.9998653621757473` (only 3 predictions wrong) using libsvm-based `SVC` class on sklearn. The ThunderSVM model yields only 50 support vectors in my test. The libSVM model yields 21195 support vectors. Note: No extra parameters were passed to the `SVC()` constructor in either case.

Caveat: admittedly the dataset has NOT been normalized. That could contribute to the extreme error here.

This could be related to the previous issue I submitted (https://github.com/Xtra-Computing/thundersvm/issues/125) but I am not sure, so this is filed into a separate issue.

[train.svm.gz](https://github.com/Xtra-Computing/thundersvm/files/2860282/train.svm.gz)

**Comment by zeyiwen** [comment]
Thanks for the feedback! It doesn't seem to be related to #125. We will investigate it, and fix this issue as soon as possible.

**Comment by zeyiwen** [comment]
@wirawan0 Hi there. We have fixed the issue. You need to use ``-DUSE_DOUBLE=ON`` when runing the ``cmake`` command. We have obtain an accuracy similar to LibSVM's.

The issue is related to precision again, because GPUs have better efficiency on single precision floating-point values. The ``float`` type works fine for many problems. It doesn't work in your problem, because the data is not normalized. Hence, you need to increase the precision to ``double``.

**Comment by zeyiwen** [comment]
The issue should be fixed now, and hence we like to close it.
